sklearn/preprocessing/imputation.py

```diff
@@ -311,37 +326,48 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
-    def _sparse_transform(self, X, valid_stats, valid_idx):
+    def _sparse_transform(self, X, valid_stats):
         """transformer on sparse data."""
         mask = _get_mask(X.data, self.missing_values)
         indexes = np.repeat(np.arange(len(X.indptr) - 1, dtype=np.int),
                             np.diff(X.indptr))[mask]
+        if self.axis == 0:
+            not_mask = np.logical_not(mask)
```

Comment:
Rather than duplicating this data, can't you just check for zero elements over the sum?